internal/controllers/machinedeployment/machinedeployment_rollout_rollingupdate.go

```diff
@@ -365,6 +372,11 @@ func (p *rolloutPlanner) scaleDownOldMSs(ctx context.Context, totalScaleDownCoun
 
 		if scaleDown > 0 {
 			newScaleIntent := max(replicas-scaleDown, 0)
+			if oldTotalAvailableReplicas == totalAvailableReplicas {
```

Comment:
Please add a godoc explaining why we can ~ infer these notes from these checks

Also for additional oldMS the oldTotalAvailableReplicas number in l.378 will be just entirely wrong, right?